src/Tizen.Nlp/Tizen.Nlp/NaturalLanguageProcess.cs

```diff
@@ -0,0 +1,251 @@
+    {
+        /// <summary>
+        /// An EventHandler to receive data after wordtokenize processing.
+        /// </summary>
+        /// <remarks>
+        /// if this succeeds, the object contain Key/Pari about Nlp will be received.
+        /// and you can use TryGetValue() with the Key "token" or "tag" to get a array of string for access,
+        /// </remarks>
+        public event EventHandler<MessageReceivedEventArgs> WordTokenizeResultReceived;
+        /// <summary>
+        /// An EventHandler to recieve data after the pos of tag processing.
+        /// </summary>
+        /// <remarks>
+        /// if this succeeds, the object contain Key/Pari about Nlp will be received.
+        /// and you can use TryGetValue() with the Key "token" or "tag" to get a array of string for access,
+        /// </remarks>
+        public event EventHandler<MessageReceivedEventArgs> PosTagResultReceived;
+        /// <summary>
+        /// An EventHandler to recieve data after named entity recognition processing.
+        /// </summary>
+        /// <remarks>
+        /// if this succeeds, the object contain Key/Pari about Nlp will be received.
+        /// and you can use TryGetValue() with the Key "token" or "tag" to get a array of string for access,
+        /// </remarks>
+        public event EventHandler<MessageReceivedEventArgs> NamedEntityRecognitionResultReceived;
+        /// <summary>
+        /// An EventHandler to recieve data after lemmatize processing.
+        /// </summary>
+        /// <remarks>
+        /// if this succeeds, the object contain Key/Pari about Nlp will be received.
+        /// and you can use TryGetValue() with the Key "token" or "tag" to get a array of string for access,
+        /// </remarks>
+        public event EventHandler<MessageReceivedEventArgs> LemmatizeResultReceived;
+        /// <summary>
+        /// An EventHandler to recieve data after language detect processing.
+        /// </summary>
+        /// <remarks>
+        /// if this succeeds, the object contain Key/Pari about Nlp will be received.
+        /// and you can use TryGetValue() with the Key "token" or "tag" to get a array of string for access,
+        /// </remarks>
+        public event EventHandler<MessageReceivedEventArgs> LanguageDetectResultReceived;
+        private Message _msg;
+        private readonly Message.NotifyCb _noti = new Message.NotifyCb();
+        private string _tag;
+        private const string SERVICE_ID = "org.tizen.nlp.service";
+
+        /// <summary>
+        /// The enum type of the message be sent.
+        /// </summary>
+        public enum ProcessResult
+        {
+            /// <summary>
+            ///  Unsuccess
+            /// </summary>
+            ProcessError = -1,
+            /// <summary>
+            /// Success
+            /// </summary>
+            ProcessSuccess = 0
+        }
+
+        private ProcessResult MakeRequest(string cmd, string sentence)
```

Comment:
X DO NOT return error codes. 
https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/exception-throwing

Reply:
You need to distinguish each request to mapping result of request